crates/xilem_core/src/any_view.rs

```diff
@@ -3,22 +3,22 @@
 
 #[macro_export]
 macro_rules! generate_anyview_trait {
-    ($anyview:ident, $viewtrait:ident, $viewmarker:ty, $cx:ty, $changeflags:ty, $anywidget:ident, $boxedview:ident; $($ss:tt)*) => {
+    ($anyview:ident, $viewtrait:ident, $viewmarker:ty, $cx:ty, $changeflags:ty, $anywidget:ident, $boxedview:ident; ($($super_bounds:tt)*), ($($state_bounds:tt)*)) => {
         /// A trait enabling type erasure of views.
         pub trait $anyview<T, A = ()> {
             fn as_any(&self) -> &dyn std::any::Any;
 
             fn dyn_build(
                 &self,
                 cx: &mut $cx,
-            ) -> ($crate::Id, Box<dyn std::any::Any $( $ss )* >, Box<dyn $anywidget>);
+            ) -> ($crate::Id, Box<dyn std::any::Any $( $state_bounds )* >, Box<dyn $anywidget>);
```

Comment:
You could add the `+` here and the following uses of `$state_bounds` and remove it from `generate_anyview_trait!` invocations so they look less weird. A trailing `+` in a list of trait bounds is not a syntax error.

Reply:
Well that's interesting (same with the `:`), new knowledge acquired...

Reply:
You can also have an empty `where` clause, or empty generic parameter list (`<>`) behind non-generic types *including primitives* 😄

Though if you use rustfmt, it will remove all of these things.